### PR TITLE
Media reporting e2e tests, single audio support, style updates, and accessibility improvements

### PR DESCRIPTION
**Pull request by zackkrida**
## Description

This PR makes a number of improvements to the reporting media code:

- Updates the styles of the reporting popup trigger button to match the mockups
- Adds the reporting functionality to single audio results
- Updates the report form to use an actual submit button and form handler, rather than a click event on a `type="button"` button.
- Updates the reporting API service to require a media type
- Passes the `media.frontendMediaType` property to the reporting service to make the form work with all media tyes
- Adds e2e tests for each report type for each media type

I would love some heavy attention applied to the e2e tests. These tests run each type of report (dmca, mature, and other) on the first search result each media-specific search endpoint (`search/images`, for example). This tests the entire flow, like this:

1. Choose the desired media type from the content switcher
2. Search for 'cat'
3. Click on the first result.
4. Open the report modal.
5. Submit the current report (there's one test per media type per report)

Currently the tests mock the various network requests made by the reporting form: the google docs form and the api reporting endpoint. There are also likely utilities in this test that could be used elsewhere, if you have any suggestions for those. 

## Testing Instructions

- Report some images and audio locally
- Run the e2e tests locally
- View the reporting button styles in 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
Side note: the VS Code Playwright extension is _amazing_! fun example, it understands and lets me run these 'dynamic' tests, it's magical:

CleanShot 2022-04-14 at 16 13 11@2x

File: src/components/VAudioDetails/VAudioDetails.vue [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
   <section class="audio-info">
-    <h4 class="text-base lg:text-3xl mb-6">
-      {{ $t('audio-details.information') }}
-    </h4>
+    <header class="flex flex-row justify-between items-center mb-6">
+      <h4 class="text-base lg:text-3xl">
+        {{ $t('audio-details.information') }}
+      </h4>
+      <VContentReportPopover :media="audio" />
+    </header>
```
Comment (zackkrida) [inline src/components/VAudioDetails/VAudioDetails.vue:8]: This adds the report form and button to the single audio results.

File: src/components/VContentReport/VContentReportButton.vue [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
     variant="plain"
-    class="report-button font-semibold text-dark-charcoal"
+    class="report-button font-semibold text-dark-charcoal-70"
   >
-    <span class="md:hidden">{{
+    <span class="text-sr md:text-base md:hidden">{{
       $t('media-details.content-report.short')
     }}</span>
-    <span class="hidden md:inline">{{
+    <span class="text-sr md:text-base hidden md:inline">{{
```
Comment (zackkrida) [inline src/components/VContentReport/VContentReportButton.vue:9]: Style updates to the reporting button so it matches the mockups:

CleanShot 2022-04-14 at 16 16 24@2x

File: src/components/VContentReport/VContentReportForm.vue [hunk 3]
```diff
@@ -93,10 +93,11 @@
           <VButton
             v-else
             key="non-dmca"
+            type="submit"
             :disabled="isSubmitDisabled"
             :focusable-when-disabled="true"
             variant="secondary"
-            @click="handleSubmit"
+            :value="$t('media-details.content-report.form.submit')"
```
Comment (zackkrida) [inline src/components/VContentReport/VContentReportForm.vue:100]: Fun fact: Playwright's selectors for text `page.locate('text="Submit"'), for example, use the `value` when applied to submit buttons, not the actual button text contents.

File: src/components/VContentReport/VContentReportForm.vue [hunk 4]
```diff
@@ -166,11 +167,13 @@ export default defineComponent({
     const isSubmitDisabled = computed(
       () => selectedReason.value === OTHER && description.value.length < 20
     )
-    const handleSubmit = async () => {
+    const handleSubmit = async (event) => {
+      event.preventDefault()
```
Comment (zackkrida) [inline src/components/VContentReport/VContentReportForm.vue:171]:
```suggestion
      event.preventDefault()
```

Prevents redirection to the api POST request.

File: test/playwright/e2e/report-media.spec.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  await newPage.waitForLoadState()
+  return expect(await newPage.url()).toContain('https://docs.google.com/forms')
+}
+
+// todo: Test a mature report with the optional description field
+const submitMatureContentReport = async (
+  page: Page,
+  context: BrowserContext
+) => {
+  const reportingEndpoint = /https:\/\/api.openverse.engineering\/\w+\/\w+/
```
Comment (zackkrida) [inline test/playwright/e2e/report-media.spec.ts:37]: Oops, need to fix this. The whole reason I used a regex in the first place was to match `https://api.openverse.engineering/{mediatype}/{id}/report/`

**Comment by zackkrida** [comment]
Hm, it looks like an unrelated e2e test is failing. Should I fiddle with the tapes or something?

**Comment by sarayourfriend** [comment]
> I imagine I should also mock the search responses?

What do you mean by this? As long as there are tapes for the search requests then they are mocked.

File: test/playwright/e2e/report-media.spec.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  dmca: submitDmcaReport,
+  mature: submitMatureContentReport,
+  other: submitOtherReport,
+}
+
+/**
+ * Iterate through all the media types and supported reports
+ * to make sure every permutation works correctly.
+ */
+mediaTypes.map((mediaType) => {
```
Comment (sarayourfriend) [inline test/playwright/e2e/report-media.spec.ts:92]: nit (I mean _super_ nit-picky): To me `map` implies that the return value of the iteration will be useful, but here it's just a plain `for` loop.

```suggestion
mediaTypes.forEach((mediaType) => {
```

**Review by sarayourfriend — COMMENTED** [review]
Just reviewing the code so far. E2e tests look good to me!

File: src/components/VContentReport/VContentReportForm.vue (hunk 3 above)
Comment (obulat) [reply src/components/VContentReport/VContentReportForm.vue:100]: TIL! I guess we should add `value` attribute to all the submit buttons? I wonder if it uses `value` for all buttons...

**Review by sarayourfriend — APPROVED** [review]
LGTM!

**Comment by zackkrida** [comment]
> I imagine I should also mock the search responses?

@sarayourfriend I wrote this before I fully understood the tapes 😅 disregard!

File: src/components/VContentReport/VContentReportForm.vue [hunk 7]
```diff
@@ -166,11 +167,13 @@ export default defineComponent({
...
       if (selectedReason.value === DMCA) return
       // Submit report
       try {
         await service.sendReport({
+          mediaType: props.media.frontendMediaType,
```
Comment (obulat) [inline src/components/VContentReport/VContentReportForm.vue:176]: I keep forgetting that we can use the `frontendMediaType` property!

**Review by obulat — APPROVED** [review]
Great improvements in the way the report sets the mediaType, and overall report style and flow.
I wonder why I can't see the CI run information?

**Pull request merged**